test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/MvcEndpointDataSourceTests.cs

```diff
@@ -606,18 +617,37 @@ public void RequiredValues_NotPresent_InDefaultValues_IsAddedToDefaultValues()
             var actionDescriptorCollection = GetActionDescriptorCollection(requiredValues: requiredValues);
             var dataSource = CreateMvcEndpointDataSource(actionDescriptorCollection);
             dataSource.ConventionalEndpointInfos.Add(
-                CreateEndpointInfo(string.Empty, "{controller=Home}/{action=Index}"));
+                CreateEndpointInfo(string.Empty, "{subarea}/{controller=Home}/{action=Index}"));
 
             // Act
             var endpoints = dataSource.Endpoints;
 
             // Assert
             var endpoint = Assert.Single(endpoints);
             var matcherEndpoint = Assert.IsType<RouteEndpoint>(endpoint);
-            Assert.Equal("Foo/Bar", matcherEndpoint.RoutePattern.RawText);
+            Assert.Equal("test/Foo/Bar", matcherEndpoint.RoutePattern.RawText);
             AssertIsSubset(expectedDefaults, matcherEndpoint.RoutePattern.Defaults);
         }
 
+        [Fact]
+        public void RequiredValues_NotPresent_InDefaultValuesOrParameter_EndpointNotCreated()
```

Comment:
I didn't see good coverage for the fact that defaults behave totally differerntly when they are not parameters